roachprod: TBD (--dry-run, --verbose, logger refactor, RAID0 refactor)

srosenberg opened the pull request:
WIP

Release note: None

srosenberg committed:
roachprod: TBD (--dry-run, --verbose, logger refactor, RAID0 refactor)

WIP

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/100966)
<!-- Reviewable:end -->

srosenberg committed:
roachtest: add optional PreSetup and PostSetup hooks

TBD: summarize

Epic: none

Release note: None

Resolves: https://github.com/cockroachdb/cockroach/issues/33377

srosenberg committed:
roachtest: add optional PreSetup and PostSetup hooks

TBD: summarize

Epic: none

Release note: None

Resolves: https://github.com/cockroachdb/cockroach/issues/33377

srosenberg committed:
roachtest: add optional PreSetup and PostSetup hooks

TBD: summarize

Epic: none

Release note: None

Resolves: https://github.com/cockroachdb/cockroach/issues/33377
